lib/modules/datasource/aws-eks-addon/index.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  type AddonInfo,
+  type AddonVersionInfo,
+  type Compatibility,
+  DescribeAddonVersionsCommand,
+  type DescribeAddonVersionsCommandInput,
+  type DescribeAddonVersionsCommandOutput,
+  EKSClient,
+} from '@aws-sdk/client-eks';
+import { fromNodeProviderChain } from '@aws-sdk/credential-providers';
+import { logger } from '../../../logger';
+import { cache } from '../../../util/cache/package/decorator';
+// import * as awsEksAddonVersioning from '../../versioning/aws-eks-addon';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import { EksAddonsFilter } from './schema';
+
+export class AwsEKSAddonDataSource extends Datasource {
+  static readonly id = 'aws-eks-addon';
+
+  // override readonly defaultVersioning = awsEksAddonVersioning.id;
```

Comment:
why not using the eks addon versioning?
```suggestion
  override readonly defaultVersioning = awsEksAddonVersioning.id;
```